src/main/kotlin/lotto/domain/AutoStrategy.kt

```diff
@@ -0,0 +1,26 @@
+package lotto.domain
+
+class AutoStrategy : LottoStrategy {
+    override fun generateLotto(money: Money): Lottoes {
+        val quantity = (money.currentMoney / LottoGame.LOTTO_COST).toInt()
+        money.spendAllMoney()
+        return Lottoes(
+            (1..quantity).map {
+                makeAutoLottoTicket()
+            }
+        )
+    }
+
+    private fun makeAutoLottoTicket(): LottoTicket {
+        val numbers = (LottoNumber.MIN_LOTTO_NUMBER..LottoNumber.MAX_LOTTO_NUMBER)
+            .shuffled()
+            .take(LottoTicket.LENGTH_OF_LOTTO)
+            .sorted()
+
+        return LottoTicket(
+            numbers.map { number ->
+                LottoNumber.from(number)
+            }.toSet()
+        )
```

Comment:
생성자의 수가 늘어날수록 클래스를 사용하기 위한 유연성은 높아집니다.

```suggestion
        return LottoTicket(numbers)
```